server/src/main/java/org/elasticsearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -1077,6 +1079,9 @@ public void validateIndexSettings(String indexName, final Settings settings, fin
      */
     static List<String> validateShrinkIndex(ClusterState state, String sourceIndex, String targetIndexName, Settings targetIndexSettings) {
         IndexMetadata sourceMetadata = validateResize(state, sourceIndex, targetIndexName, targetIndexSettings);
+        if ("snapshot".equals(INDEX_STORE_TYPE_SETTING.get(sourceMetadata.getSettings()))) {
```

Comment:
Could we move `SearchableSnapshotsConstants#SNAPSHOT_DIRECTORY_FACTORY_KEY` and `SearchableSnapshotsConstants#isSearchableSnapshotStore` to `server` rather than just using the literal `"snapshot"` throughout? Either that or add an independent constant in `server` and then add a static constructor to `o.e.x.c.searchablesnapshots.SearchableSnapshotsConstants` to assert that the constant in `server` is equal to `SNAPSHOT_DIRECTORY_FACTORY_KEY`?

Reply:
Yes, I'll do something like this in a follow-up (it touches places outside of this pull request too)